behind-the-veil-siteroot/imports/ui/components/searchBar/searchBar.jsx

```diff
@@ -4,57 +4,244 @@
+                    return matchMain || matchSub;
+                });
+                setFilteredSuggestions(filteredSuggestions);
+            }
+        } else {
+            setFilteredSuggestions([]);
+        }
+    }, [inputValue])
+
+    // on search type change
+    useEffect(() => {
+        setFullSuggestions(useSearchSuggestions(searchType, usersData, servicesData)
+            .sort((a, b) => a.main.localeCompare(b.main)));
+        setFilteredSuggestions([]);
+    }, [searchType])
+
+
+    // search type input handler
+    const handleSearchTypeChange = (event) => {
+        setSearchType(event.target.value);
+    };
 
     // Function that resets the input value of the search bar
     const handleReset = () => {
-        onChange({target: {value: ''}});
+        setInputValue('');
+        setFilteredSuggestions([]);
     };
 
-    return (
-        // The form which holds the search input field and the reset button
-        <form className="flex h-12" onSubmit={handleSubmit}>
-            {/* The search input field (i.e. the search bar) */}
-            <Input
-                {...searchBarProps}
-                type="search"
-                className={classNames("rounded-r-none border-r-0 w-[250px] sm:w-[35vw] ", className)}
-                placeholder={placeholderName}
-                value={value}
-                onChange={onChange}
-            />
-            {/* The reset button which resets the value of the search input field to an empty string */}
-            <button type="button"
-                    className={"input-base flex justify-center items-center w-12 border-l-0 rounded-l-none"}
-                    onClick={handleReset}>
-                <XMarkIcon className="size-7 stroke-2 rounded-full p-1
-                hover:bg-white-hover active:bg-light-grey-hover transition duration-200 ease-in-out"></XMarkIcon>
-            </button>
-        </form>
-    );
+    // when a suggestion is selected, use the value of the suggestion as the input, to trigger the use effect that trigger
+    // on input change above
+    const handleSuggestionSelect = (value) => {
+        setInputValue(value.main);
+        setSelectedSuggestions(true) // this ensures code goes through a different logic branch than for normal input change
+        setFilteredSuggestions([]);
+    };
+
+    // when finished loading data
+    if (!isLoadingServices && !isLoadingArtists) {
+
+        // load initial full suggestions the first time page loads, and sort it
+        if (!loadedFullSuggestions) {
+            setFullSuggestions(useSearchSuggestions(searchType, usersData, servicesData)
+                .sort((a, b) => a.main.localeCompare(b.main)));
+            setLoadedFullSuggestions(true)
+        }
+
+        // load the css of the suggestions depending on if it goes UP or down
+        let ulClassnames = "w-[246px] sm:w-[calc(35vw+48px)] z-[29] absolute flex"
```

Comment:
Do these have to be hardcoded? Just wondering where the values come from

Reply:
width of the search bar add 48px for the clear button

![image](https://github.com/user-attachments/assets/811b92dc-8c28-4264-a69a-dd43dd102361)
![image](https://github.com/user-attachments/assets/38158db6-4384-4320-ba27-785ab851cd14)

Reply:
because its absolute positioning, it needs hardcoding, I cant do "width-full" or anything. Using relative positioning doesn't work (at least for me when i started with it)